cpp/ql/test/library-tests/ir/range-analysis/SimpleRangeAnalysis_tests.cpp

```diff
@@ -997,3 +997,15 @@ void test_overflow() {
     range(x + y); // $ range===-2147483393
   }
 }
+
+void test_negate_unsigned(unsigned u) {
+  if(10 < u && u < 20) {
+    range<unsigned>(-u); // underflows
```

Comment:
I assume that "underflows" here refers to how we're reasoning about ranges, i.e., not taking into account any wrap around of the type?

Reply:
That was my intention, yes. We could infer a very large upper and lower bound for these, but these bounds wouldn't be helpful. So instead we just infer no bound. Hopefully the overflow detection @rdmarsh is working on will be able to conclude that there's a chance of underflow here.